cronitor/celery.py

```diff
@@ -0,0 +1,147 @@
+import typing
+import datetime
+import humanize
+import logging
+from cronitor import State, Monitor
+import cronitor
+import sys
+
+logger = logging.getLogger(__name__)
+try:
+    import celery
+    import celery.beat
+    from celery.schedules import crontab, schedule, solar
+    from celery.signals import beat_init, task_prerun, task_failure, task_success, task_retry
+
+    if typing.TYPE_CHECKING:
+        from typing import Dict, List, Union, Optional
+        import billiard.einfo
+        from celery.worker.request import Request
+except ImportError:
+    logger.error("Cannot use the cronitor.celery module without celery installed")
+    sys.exit(1)
+
+# For the signals to properly register, they need to be top-level objects.
+# Since they are defined dynamically in initialize(), we have to declare them up top,
+# make them global, and override them.
+celerybeat_startup = None
+ping_monitor_before_task = None
+ping_monitor_on_success = None
+ping_monitor_on_failure = None
+ping_monitor_on_retry = None
+
+
+def get_headers_from_task(task):  # type: (celery.Task) -> Dict
+    headers = task.request.headers or {}
+    headers.update(task.request.properties.get('application_headers', {}))
+    return headers
+
+
+def initialize(app, api_key=None):  # type: (celery.Celery, Optional[str]) -> None
+    if api_key:
+        cronitor.api_key = api_key
+
+    global celerybeat_startup
```

Comment:
should we prefix these with `cronitor_` if they are exposed as globals?

Reply:
AFAIK it's "global" only to the module. It shouldn't leak outside the file

Reply:
I was trying to remember my Python global rules and couldn't :)